src/effects/deeplearning/ModelManagerPanel.cpp

```diff
@@ -0,0 +1,809 @@
+      {
+         if (pair.first == card->GetRepoID())
+            pair.second->SetModelStatus(ModelCardPanel::ModelStatus::enabled);
+      }
+   }
+
+   // configure the detailed panel
+   if (card)
+   {
+      mDetailedPanel->PopulateWithNewCard(card);
+      mDetailedPanel->SetModelStatus(ModelCardPanel::ModelStatus::enabled);
+   }
+
+}
+
+// ManagerToolsPanel
+
+ManagerToolsPanel::ManagerToolsPanel(wxWindow *parent, ModelManagerPanel *panel)
+   : wxPanelWrapper((wxWindow *)parent, wxID_ANY, wxDefaultPosition, wxSize(MANAGERPANEL_WIDTH, 30))
+{
+   mManagerPanel = panel;
+   mFetchStatus = nullptr;
+   mAddRepoButton = nullptr;
+   mExploreButton = nullptr;
+
+   ShuttleGui S(this, eIsCreating);
+   PopulateOrExchange(S);
+
+   SetWindowStyle(wxBORDER_SIMPLE);
+   // Fit();
+   Layout();
+   // Center();
+   // SetMinSize(GetSize());
+   Refresh();
+}
+
+void ManagerToolsPanel::PopulateOrExchange(ShuttleGui &S)
+{
+   S.StartHorizontalLay(wxLEFT, true);
+   {
+      mAddRepoButton = S.AddButton(XO("Add From HuggingFace"));
+      mExploreButton = S.AddButton(XO("Explore Models"));
+      mFetchStatus = S.AddVariableText(XO("Fetching models..."), 
+                                 true, wxALIGN_CENTER_VERTICAL);
+   }
+   S.EndHorizontalLay();
+
+   mAddRepoButton->Bind(wxEVT_BUTTON, &ManagerToolsPanel::OnAddRepo, this);
+   mExploreButton->Bind(wxEVT_BUTTON, &ManagerToolsPanel::OnExplore, this);
+}
+
+void ManagerToolsPanel::OnAddRepo(wxCommandEvent & WXUNUSED(event))
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   CardFetchedCallback onCardFetched = mManagerPanel->GetCardFetchedCallback();
+
+   wxString msg = XO("Enter a HuggingFace Repo ID \n"
+                    "For example: \"huggof/ConvTasNet-DAMP-Vocals\"\n").Translation();
+   wxString caption = XO("AddRepo").Translation();
+   wxTextEntryDialog dialog(this, msg, caption, wxEmptyString);
+
+   if (dialog.ShowModal() == wxID_OK)
+   {
+      std::string repoId = dialog.GetValue().ToStdString();
+      manager.FetchCard(repoId, onCardFetched);
+   }
+}
+
+void ManagerToolsPanel::SetFetchProgress(int64_t current, int64_t total)
+{
+   if (!mFetchStatus)
+      return;
+
+   if (total == 0)
+   {
+      wxString translated = XO("Error fetching models.").Translation();
+      mFetchStatus->SetLabel(translated);
+   }
+
+   if (current < total)
+   {
+      // TODO: this should be a translatable string
+      wxString translated = XO("Fetching %d out of %d")
+            .Format((int)current, (int)total).Translation();
+      mFetchStatus->SetLabel(translated);
+   }
+
+   if (current == total)
+   {
+      mFetchStatus->SetLabel(XO("Manager ready.").Translation());
+   }
+}
+
+void ManagerToolsPanel::OnExplore(wxCommandEvent & WXUNUSED(event))
+{
+   ExploreDialog dialog = ExploreDialog(mManagerPanel->GetParent(), mManagerPanel);
+   dialog.ShowModal();
+}
+
+// ExploreDialog
+
+ExploreDialog::ExploreDialog(wxWindow *parent, ModelManagerPanel *panel)
```

Comment:
More explicit name for the class.

And let's move it to a different set of h/cpp

Reply:
https://github.com/audacity/audacity/pull/1384/commits/c35bf2320839b284103a5cbbbc34bd76a301fa39